packit_service/worker/parser.py

```diff
@@ -722,52 +727,77 @@ def parse_distgit_event(event) -> Optional[DistGitEvent]:
         )
 
     @staticmethod
-    def parse_testing_farm_results_event(event) -> Optional[TestingFarmResultsEvent]:
-        """ this corresponds to testing farm results event """
-        pipeline_id: str = nested_get(event, "pipeline", "id")
-        if not pipeline_id:
-            return None
-
-        result: TestingFarmResult = TestingFarmResult(event.get("result"))
-        environment: str = nested_get(event, "environment", "image")
-        message: str = event.get("message")
-        log_url: str = event.get("url")
-        copr_repo_name: str = nested_get(event, "artifact", "copr-repo-name")
-        copr_chroot: str = nested_get(event, "artifact", "copr-chroot")
-        repo_name: str = nested_get(event, "artifact", "repo-name")
-        repo_namespace: str = nested_get(event, "artifact", "repo-namespace")
-        ref: str = nested_get(event, "artifact", "git-ref")
-        https_url: str = nested_get(event, "artifact", "git-url")
-        commit_sha: str = nested_get(event, "artifact", "commit-sha")
-        tests: List[TestResult] = [
+    def _parse_tf_result_xunit(xunit: Optional[str]) -> List[TestResult]:
+        """ Parse event["result"]["xunit"] to get tests results """
+        if not xunit:
+            return []
+        xunit_dict = xmltodict.parse(xunit)
+        return [
             TestResult(
-                name=raw_test["name"],
-                result=TestingFarmResult(raw_test["result"]),
-                log_url=raw_test.get("log"),
+                name=testcase["@name"],
+                result=TestingFarmResult(testcase["@result"]),
+                log_url=nested_get(testcase, "logs", "log", 1, "@href", default=""),
+            )
+            for testcase in nested_get(
+                xunit_dict, "testsuites", "testsuite", "testcase", default=[]
             )
-            for raw_test in event.get("tests", [])
         ]
 
-        logger.info(f"Results from Testing farm event. Pipeline ID: {pipeline_id}")
+    @staticmethod
+    def parse_testing_farm_results_event(
+        event: dict,
+    ) -> Optional[TestingFarmResultsEvent]:
+        """ this corresponds to testing farm results event """
+        logger.debug(f"parse_testing_farm_results_event: {event}")
+        if event.get("source") != "testing-farm" or not event.get("request_id"):
+            return None
+
+        request_id: str = event["request_id"]
+        logger.info(f"Testing farm notification event. Request ID: {request_id}")
+
+        # Testing Farm sends only request/pipeline id in a notification.
+        # We need to get more details ourselves.
+        # It'd be much better to do this in TestingFarmResultsHandler.run(),
+        # but all the code along the way to get there expects we already know the details.
+        event = TestingFarmJobHelper.get_request_details(request_id)
```

Comment:
What is the downside of calculating the missing attributes here? That we don't want to do it in the service?

I think we do some calculating also for other events.

But we can optimize later.

Reply:
Not sure I understand the question.

I don't see how we could calculate anything else from just a request_id, which is all we get in a notification from TF.

Example: When something changes in the TF request, TF sends a notification that contains only request id (for example `ab18144c-777b-4b54-aaea-a8f32f1c3d59`) and we have to get https://api.dev.testing-farm.io/v0.1/requests/ab18144c-777b-4b54-aaea-a8f32f1c3d59 to know all the other attributes.

There are 3 places when we can do this (get request details from TF):
- in service (httpd): we don't want to block service when something's wrong with TF and we want to have some retry mechanism
- in task.run_testing_farm_results_handler: that's what I wanted initially, but it didn't go well, hence the comment
- here in task.steve_jobs.process_message

Reply:
Ok, thanks for the info. I now understand that more.

Last question:
- Do we have the info about test-run already in the database? (Added when we trigger the test?)

Reply:
Hopefully, I answer your question by saying that yes, we [create a TFTTestRunModel at the end of TestingFarmJobHelper.run_testing_farm()](https://github.com/packit/packit-service/pull/875/files#diff-bdd081387e0aa002e9e49b42836616251db4ab0822b9d6d47dfbdc5c0c0e6a46R218)

Reply:
So, what about getting the missing information from the database?

If we know the pipeline-id, we have a relation to the project and trigger model.

Reply:
Aha, I see now. I've been always weak in these databazy thingy.
Look, I don't feel like doing it right now. I'd rather merge this ASAP and do this change later - I've added a TODO comment there.